docs/sources/operations/query-acceleration-blooms.md

```diff
@@ -1,48 +1,66 @@
+weight:
 keywords:
   - blooms
   - query acceleration
 ---
 
 # Query Acceleration with Blooms (Experimental)
+
 {{% admonition type="warning" %}}
-This feature is an [experimental feature](/docs/release-life-cycle/). Engineering and on-call support is not available.  No SLA is provided.  
+This feature is an [experimental feature](/docs/release-life-cycle/). Engineering and on-call support is not available. No SLA is provided.
 {{% /admonition %}}
 
-Loki 3.0 leverages [bloom filters](https://en.wikipedia.org/wiki/Bloom_filter) to speed up queries by reducing the 
-amount of data Loki needs to load from the store and iterate through. Loki is often used to run “needle in a haystack” 
-queries; these are queries where a large number of log lines are searched, but only a few log lines match the [filtering 
-expressions]({{< relref "../query/log_queries#line-filter-expression" >}}) of the query. 
-Some common use cases are needing to find a specific text pattern in a message, or all logs tied to a specific customer ID.
+Loki leverages [bloom filters](https://en.wikipedia.org/wiki/Bloom_filter) to speed up queries by reducing the amount of data Loki needs to load from the store and iterate through.
+Loki is often used to run "needle in a haystack" queries; these are queries where a large number of log lines are searched, but only a few log lines match the query.
+Some common use cases are needing to find all logs tied to a specific trace ID or customer ID.
 
 An example of such queries would be looking for a trace ID on a whole cluster for the past 24 hours:
 
 ```logql
-{cluster="prod"} |= "traceID=3c0e3dcd33e7"
+{cluster="prod"} | traceID="3c0e3dcd33e7"
 ```
 
-Loki would download all the chunks for all the streams matching `{cluster=”prod”}` for the last 24 hours and iterate
-through each log line in the chunks checking if the string `traceID=3c0e3dcd33e7` is present.
+Without accelerated filtering, Loki downloads all the chunks for all the streams matching `{cluster="prod"}` for the last 24 hours and iterates through each log line in the chunks, checking if the [structured metadata][] key `traceID` with value `3c0e3dcd33e7` is present.
+
+With accelerated filtering, Loki is able to skip most of the chunks and only process the ones where we have a statistical confidence that the structured metadata pair might be present.
+
+## Using query acceleration
+
+### Add data to blooms
+
+To make data available for query acceleration, send [structured metadata][] to Loki. Loki builds blooms from all strucutred metadata keys and values.
 
-With accelerated filtering, Loki is able to skip most of the chunks and only process the ones where we have a 
-statistical confidence that the string might be present. 
-The underlying blooms are built by the [Bloom Builder](#bloom-planner-and-builder) component
-and served by the new [Bloom Gateway](#bloom-gateway) component.
+### Query blooms
```

Comment:
I would put the query section AFTER the section about how to enable blooms.

Reply:
Hm, can we discuss options here? 

It's likely that there are going to be many more people going to this page who aren't Loki operators but need to know the rules for using blooms, like a cloud user or someone on a team that uses Loki but does not operate it themselves. I'm concerned that moving information for how to take advantage of blooms after all the operator content would hurt discoverability. 

Do you think that's a risk? Would there be another way to organize the content?

Reply:
Well, you can't really query blooms if they're not enabled yet....

Reply:
Right, but the audience using blooms and the audience managing advantage of blooms are sometimes distinct. How can we educate people on how to write queries which take advantage of the enabled blooms without forcing them to scan through information about how to enable them? 

Or is this problem not worth solving? 

(For example: if I was a Grafana Cloud user, only the information about sending structured metadata and how to write queries to use blooms is relevant to me)

Reply:
Maybe we need to split the information out about querying blooms and put it in a topic under [Query](https://grafana.com/docs/loki/latest/query/) instead?

Reply:
@JStickler Good idea, thanks! I tried that in my latest commit, LMK what you think.